ICRCs/ICRC-107/ICRC-107.md

```diff
@@ -0,0 +1,188 @@
+
+Fee collection settings determine how fees are processed. These settings include:
+
+- A fee collector account (`fee_col`). If `fee_col` is not set, all fees are burned.
+- A list of operations (`col_ops`) for which fees are collected. If an operation is not in `col_ops`, the fee is burned.
+
+Changes to fee collection settings are recorded on-chain, ensuring a transparent history of modifications.
+
+### 2.2 Fee Collection Settings
+
+**Backwards Compatibility:**
+To ensure compatibility with existing ICRC-3 implementations, `fee_col` **MUST** continue to be recorded using the existing array format:
+
+```
+fee_col;
+variant {
+  Array = vec {
+    variant { Blob = principal_blob };
+    variant { Blob = subaccount_blob };
+  }
+}
+```
+
+- The **first Blob** represents the **principal** of the fee collector.
+- The **second Blob** represents the **subaccount**, or a zeroed-out Blob if no subaccount is used.
+
+This ensures that existing tools and explorers relying on the current `fee_col` format remain functional.
+
+A block **MAY** include the following fields to define or update fee collection settings:
+
+- **`fee_col`** (optional, `Map<Value>`)
+
+  - `principal`: `Blob` — The principal of the fee collector.
+  - `subaccount`: `Blob` (optional) — The subaccount identifier, if applicable.
+
+- **`col_ops`** (optional, `Vec<Text>`)
+
+  - A list of operation types for which fees **should** be collected.
+  - If omitted, defaults to `"transfer"` (fees are collected only for transfers).
+
+- **`prev_fee_col_info`** (optional, `Nat`)
+
+  - The block index at which `fee_col` or `col_ops` was last updated.
+  - Enables quick lookup of previous fee collection settings.
+
+**Note:** The block format follows the [ICRC-3 standard](https://github.com/dfinity/ICRC/blob/main/ICRCs/ICRC-3.md).
+
+---
+
+## 3. Handling Fee Collection for ICRC-1 and ICRC-2 Blocks
+
+### 3.1 How to Calculate the Fee for ICRC-1 and ICRC-2 Blocks
+
+Blocks follow the ICRC-3 standard. To determine the **final fee amount** for a block:
+
+1. **Check `tx.fee`**  
+   - If present, `tx.fee` is the fee for this block.
+
+2. **Else, check the top-level `fee` field**  
+   - If `tx.fee` is **not set**, and a top-level field `fee` exists in the block, then that is the fee.
+
+3. **Else, fallback to `0`**  
+   - If **neither** `tx.fee` nor `fee` is present, the default fee is `0`.
+
+The paying account is the source account for both transfer and approve transactions.
+
+### 3.2 How `col_ops` Determines Fee Collection
+
+The `col_ops` field defines which block types incur fees that are collected instead of burned. For ICRC-1 and ICRC-2 blocks, `col_ops` entries map to block types as follows:
+
+| **col_ops Entry** | **Block Types Affected** |
+|-------------------|--------------------------|
+| **"transfer"**    | Blocks with `btype = "1xfer"` or `btype = "2xfer"`. If `btype` is not set, use `tx.op = "xfer"` or `"2xfer"`. |
```

Comment:
I'll be annoying here and ask that they be namespace. 12transfer 2approve.

Why? Inevitably someone will make an ICRC7+1 canister that will also have 7xfer blocks where fees go to creators and with 12xfer blocks where fees go to the DAO and we'll have transfer and transfer and be right back at the global namespacing conundrum.

When in doubt, namespace. We're sharing a global namespace and being specific is friendly. Good fences make good neighbors.

Reply:
Just to be clear: you do think we should consider the scenario where different fees types go to different fee collectors?

Reply:
I don't see any harm in name spacing the operations

Reply:
I've used the btype as entries in the col_ops (now called fee_ops).